src/test/java/com/cst438/controller/InstrGradesAsgnmtUploadAllScoreSystemTest.java

```diff
@@ -0,0 +1,135 @@
+        ops.addArguments("--remote-allow-origins=*");
+
+        // start the driver
+        driver = new ChromeDriver(ops);
+
+        driver.get(URL);
+        // must have a short wait to allow time for the page to download
+        Thread.sleep(SLEEP_DURATION);
+
+    }
+
+    @AfterEach
+    public void terminateDriver() {
+        if (driver != null) {
+            // quit driver
+            driver.close();
+            driver.quit();
+            driver = null;
+        }
+    }
+
+    //instructor grades an assignment and enters scores for all enrolled students
+    // and uploads the scores
+    @Test
+    public void gradeAsgnmtEnterScoresAllEnrolled() throws Exception{
+        driver.findElement(By.id("year")).sendKeys("2024");
+        driver.findElement(By.id("semester")).sendKeys("Spring");
+        Thread.sleep(SLEEP_DURATION);
+
+        WebElement we = driver.findElement(By.id("showSections"));
+        we.click();
+        Thread.sleep(SLEEP_DURATION);
+
+        try{
+            while(true){
+                //Selects Assignment in Sections
+                WebElement sectionRow = driver.findElement(By.xpath("//tr[td='Assignments']"));
+                List<WebElement> sectionLinks = sectionRow.findElements(By.tagName("a"));
+                assertEquals(2, sectionLinks.size());
+                sectionLinks.get(1).click();
+                Thread.sleep(SLEEP_DURATION);
+
+                //Selects Grade in Assignments
+                WebElement asgnmtRow = driver.findElement(By.id("sectionAssignments"));
+                List<WebElement> gradeButton = asgnmtRow.findElements(By.id("assignmentGrades"));
+                gradeButton.get(0).click();
+                Thread.sleep(SLEEP_DURATION);
+
+
+                //Selects each score field and creates a list of assignment scores
+                WebElement assignments = driver.findElement(By.id("assignments"));
+                List<WebElement> asgScores = assignments.findElements(By.name("score"));
+
+                //Asserts that the 5 text field for score exist
+                assertEquals(5, asgScores.size());
+
+                //Iterates through each score field for all students enrolled in the assignment
+                //Add score for each student and saves scores.
+                Integer add =0;
+                for(int i = 0; i < asgScores.size(); i++){
+                    Integer inputScore = (63 + 6 * i) + add;
+
+                    if(asgScores.get(i).getText().equals("")) {
+                        asgScores.get(i).clear();
+                        String inScore = inputScore.toString();
+                        asgScores.get(i).sendKeys(inScore);
+                        add=add+2;
+                        Thread.sleep(SLEEP_DURATION);
+                    }else{
+                        add=add+3;
+                        asgScores.get(i).click();
+                        Thread.sleep(SLEEP_DURATION);
+                    }
+                }
```

Comment:
Although an error should never happen with the default data set we were given, if the number of enrolled students goes over around 7-9 students, then the inputScore will exceed the databases limit of 100.

Reply:
Good catch, we should fix this... or let it go until we have failing tests because the db grew XD